src/vsr/replica.zig

```diff
@@ -8628,8 +8635,6 @@ pub fn ReplicaType(
                 assert(self.view == view_new);
                 assert(self.log_view == view_new);
                 assert(self.commit_min == self.commit_max);
-                assert(self.journal.dirty.count == 0);
-                assert(self.journal.faulty.count == 0);
```

Comment:
Hm, it is a bit sad to see these asserts go. Let's maybe extract "are there faulty slots since op_repair_min" function and assert this here and elsewhere. 

Is there some other way we can strengthen, rather than relax assertions in this PR?

Reply:
Mhmmmm, we could perhaps add this:
```C
        fn primary_assert_journal_repaired(self: *Self) bool {
            assert(self.valid_hash_chain_between(self.op_repair_min(), self.op));

            const slots = vsr.SlotRange{
                .head = self.journal.slot_for_op(self.op_repair_min()),
                .tail = self.journal.slot_with_op(self.op).?,
            };
            var iterator = self.journal.dirty.bits.iterator(.{ .kind = .set });
            while (iterator.next()) |index| {
                assert(!slots.contains(.{ .index = index }));
            }
        }
```

Reply:
Or actually, _this_ so we can call this from `repair` too!
```C
        fn primary_journal_repaired(self: *Self) bool {
            if (!self.valid_hash_chain_between(self.op_repair_min(), self.op)) return false;

            const slots = vsr.SlotRange{
                .head = self.journal.slot_for_op(self.op_repair_min()),
                .tail = self.journal.slot_with_op(self.op).?,
            };
            var iterator = self.journal.dirty.bits.iterator(.{ .kind = .set });
            while (iterator.next()) |index| {
                if (slots.contains(.{ .index = index })) return false;
            }

            return true;
        }
```

Reply:
Done in https://github.com/tigerbeetle/tigerbeetle/pull/2414/commits/1579b8f6d36250b13f6f1e0088d8102b65e3ad63!